tests/migrations/test_multidb.py

```diff
@@ -122,13 +122,16 @@ def _test_run_sql(self, app_label, should_run, hints=None):
             self.assertEqual(Pony.objects.count(), 0)
 
     @override_settings(DATABASE_ROUTERS=[MigrateNothingRouter()])
-    def test_run_sql(self):
+    def test_run_sql_migrate_nothing_router_should_not_run(self):
```

Comment:
Are these changes a necessary part of this PR or could they be done separately?

Reply:
Previous logic make table cleaning in `set_up_test_model`: https://github.com/django/django/pull/10851/files#diff-5210414c7a29d5d1f61588388b517599L54, so without table cleaning in `set_up_test_model` we have error that tables already exists, however we can run table cleaning between `_test_run_sql`, but it's a bit ugly, so I dive bit more to this tests and look like this tests cover different cases and can be splitted, another words this changes need to move cleaning tables logic to `tearDown` from `set_up_test_model`